### PR TITLE
MRG: add as_type for epochs, corresponding to issue #7365

### PR DESCRIPTION
**Pull request by SophieHerbst**
Combine mags and grads for epochs, similarly to what is done for evokeds in as_type.

### CODE REVIEW

File: mne/forward/_field_interpolation.py [hunk 1]
```diff
@@ -222,6 +223,74 @@ def _as_meg_type_evoked(evoked, ch_type='grad', mode='fast'):
...
+        data_to[ep,:,:] = np.dot(mapping, data_from[ep,:,:])
+
+    # keep only the destination channel types
+    epochs.pick_types(meg=ch_type, eeg=False, ref_meg=False)
+    
+    # this does not change the data in epochs
+    # epochs.data = data_to
+    
+    # this is a work-around
+    epochs = EpochsArray(data_to, epochs.info, tmin = epochs.times[0],
```
Comment (SophieHerbst) [inline mne/forward/_field_interpolation.py]: this is a bit of an ugly work-around, but I didn't know how to get the data back into the epochs.

**Comment by jasmainak** [comment]
It might make sense to add this to `raw` as well? maybe a mixin class?

**Comment by agramfort** [comment]
I agree that a method in InterpolationMixin or UpdateChannelsMixin is probably better. It would also avoid code duplication with code for Evoked.

@SophieHerbst it might a bit tricky to get it right. Maybe @hoechenberger can have a look this week.

**Comment by SophieHerbst** [comment]
Sure. Is it a problem for running ICA if the interpolation is applied to raw?

**Comment by hoechenberger** [comment]
> Sure. Is it a problem for running ICA if the interpolation is applied to raw?

I believe you'd have to account for the rank reduction, i.e. pass `max_pca_components=N-i`, with `N` being the original number of channels and `i` the number of interpolated channels.

**Comment by agramfort** [comment]
this one should good for review

**Comment by agramfort** [comment]
thanks @SophieHerbst and @larsoner for the team work !

**Pull request merged**